Azure/communication-ui-library-android
# [Bug] Fix crash with removal of camera permission while call ongoing

## Purpose

* Fixes a crash that occurs if a user deliberately denies video camera permission while a call is ongoing by backgrounding the app and going to App Info -> Permissions.

## Does this introduce a breaking change?

```
[ ] Yes
[x] No
```

## Pull Request Type
What kind of change does this Pull Request introduce?


```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Documentation content changes
[ ] Other... Please describe:
```

## How to Test


* Open the app and join a call
* Turn on video camera, providing permission if required
* Background the app using circle button and open up App Info using long press on our app icon
* Navigate to Permissions and turn off Camera permission
* Switch back to the (now no longer running demo app)
* The app should finish the directly launched `CallCompositeActivity` _silently_
* Re-joining the call should work and you should get a disabled Camera button

## What to Check
Verify that the following are valid
* Try turning off-and-on permissions for Camera and Microphone while a call is ongoing
* The demo app should behave reliably for permission changes, without crashing and also not cause any kind of loop between Activity starting and finishing early

## Other Information
* Added a new function `hasConfig(id:Int):Boolean` to `CallCompositeConfiguration` that allows us to check if a given instanceId is valid without causing an exception to get thrown
* `DependencyInjectionContainerHolder` now checks whether any assigned `instanceId` is present in configs and throws an `IllegalArgumentException` if it isn't found
* `CallCompositeActivity` will call `finish()` and do an early return during `onCreate()` if setting `instanceId` fails
* Checks had to be added in `onDestroy()` of `CallCompositeActivity`, `CallingFragment` and `SetupFragment` because we would still try to deallocate resources that were now not being assigned in this situation
